src/cryptonote_protocol/levin_notify.cpp

```diff
@@ -447,6 +447,8 @@ namespace levin
             next_flush = std::min(next_flush, context.flush_time);
             context.fluff_txs.reserve(context.fluff_txs.size() + txs.size());
             context.fluff_txs.insert(context.fluff_txs.end(), txs.begin(), txs.end());
+            context.fluff_txs.erase(std::unique(context.fluff_txs.begin(), context.fluff_txs.end()),
+                                    context.fluff_txs.end());
```

Comment:
I don't think the transactions are sorted at this point, although monerod will sort txs before fluffing, `fluff_txs` could contain txs from a previous fluff message, plus we aren't guaranteed that every peer will follow monerod's behavior.

I think this should be moved to just before sending: 

https://github.com/monero-project/monero/blob/cc73fe71162d564ffda8e549b79a350bca53c454/src/cryptonote_protocol/levin_notify.cpp#L396-L400

Reply:
You can see this on the previous line too - an insert is done at the end of the vector and not by sorted position.

Reply:
@vtnerd Yes. The reason why I insist to erase there is I want to prevent duplicates getting into the queue at all. (The sort has been forgotten from PR).

Although eventually what @Boog900 mentioned is better approach. Will update this PR.

Reply:
This issue fixed in new push, I am resolving this discussion. Please feel free to reopen it if you think it still applies.